.github/workflows/publish-pypi.yml

```diff
@@ -0,0 +1,106 @@
+name: Publish to PyPI
+on:
+  release:
+    types: [published]
+  push:
+    branches:
+      - main
+jobs:
+  # setup build separate from publish
+  # See https://github.com/pypa/gh-action-pypi-publish/issues/217#issuecomment-1965727093
+  build:
+    runs-on: ubuntu-latest
+    # This ensures that the publish action only runs in the main repository
+    # rather than forks
+    # Environment is encouraged so adding
+    environment: build
```

Comment:
I just created this build environment on PyPI. We will need to run a release when this is merged to test it and grab the name pyos-sphinx-theme